lec11-modelling.Rmd

```diff
@@ -0,0 +1,527 @@
+    select(term, estimate, conf.low, conf.high, p.value)
+```
+
+In this case, the uncertainty of the estimate for hindfoot length is between
+2.524 to 2.58, which is pretty narrow (that's a good thing). This tells us that
+the estimate that hindfoot length influences weight is probably reflective of
+reality. That makes sense since longer hindfeet mean there is more space and
+amount of area for more flesh to attach to and that there is more bone, which
+adds weight. We can also be more strict about our uncertainty
+(counterintuitively, a larger confidence interval is more strict).
+
+```{r}
+# Confidence interval of 99%
+tidy(fit_hf, conf.int = TRUE, conf.level = 0.99) %>% 
+    select(term, estimate, conf.low, conf.high, p.value)
+```
+
+We can also expand the model to include more terms.
+
+```{r}
+fit_hfsex <- glm(weight ~ hindfoot_length + sex, data = portal)
+tidy(fit_hfsex, conf.int = TRUE) %>% 
+    select(term, estimate, conf.low, conf.high, p.value)
+```
+
+Great! So in almost all cases, you should favour confidence intervals and
+estimation over the p-value. In some parts of science (especially biomedical
+research), p-values are ubiquitous. But, there is a major problem with p-values.
+What does the p-value *mean*?
+
+P-values are based on and used in hypothesis testing. Hypothesis testing is when
+you ask a question such as "Are males heavier than females in rodents?". There
+is then the null hypothesis ("females and males weigh the same") and the
+alternative hypothesis ("males are heavier than females"). Using statistical
+tests, we can calculate the p-value to give us an indication of which hypothesis
+is true. Traditionally, if the p-value is <0.05 you 'reject' the null
+hypothesis.
+
+However, the true meaning of the p-value is a bit counter intuitive. The
+p-value is the probability of getting a result that is the same as or more
+extreme as what you are getting given the null hypothesis is true. There is
+nothing in that statement that says the *alternate* hypothesis is true... This
+is important because often researchers use the p-value to say that their
+hypothesis is true or that the p-value is the probability their hypothesis (and
+thus reality) is true. But that is not at all the case! This is why the use of
+p-values should be minimized as much as possible and to instead use estimation
+and confidence intervals.
+
+The other thing with p-values is that they are 'unstable'/'unreliable'...
+meaning they can change with even just slight differences in the data. Let's do
+a demonstration:
+
+```{r}
+num_tests <- 100
+sample_size <- 100
+actual_difference <- 0
+variability <- 1
+
+p.values <- sapply(1:num_tests, function(x) {
+    x <- rnorm(sample_size, mean = 0, 
+               sd = variability)
+    y <- rnorm(sample_size, mean = 0 + actual_difference, 
+               sd = variability)
+    tidy(t.test(x, y))[['p.value']]
+})
+
+qplot(y = p.values) +
+    geom_hline(yintercept = 0.05)
+```
```

Comment:
Include link to dance of the p-values if they want to read more https://www.youtube.com/watch?v=5OL1RqHrZQ8

Reply:
I've got that link in the resources below! I was planning on making a suggestion to see that link for more detail